dataframe-geo/src/main/kotlin/org/jetbrains/kotlinx/dataframe/geo/WithGeometry.kt

```diff
@@ -0,0 +1,35 @@
+package org.jetbrains.kotlinx.dataframe.geo
+
+import org.jetbrains.kotlinx.dataframe.ColumnsContainer
+import org.jetbrains.kotlinx.dataframe.DataColumn
+import org.jetbrains.kotlinx.dataframe.annotations.DataSchema
+import org.locationtech.jts.geom.Geometry
+import org.locationtech.jts.geom.MultiPolygon
+import org.locationtech.jts.geom.Polygon
+
+@DataSchema
+interface WithGeometry {
+    val geometry: Geometry
+}
+
+@DataSchema
+interface WithPolygon : WithGeometry {
+    override val geometry: Polygon
+}
+
+@DataSchema
+interface WithMultiPolygon : WithGeometry {
+    override val geometry: MultiPolygon
+}
+
+@get:JvmName("geometry")
+val <T : WithGeometry> ColumnsContainer<T>.geometry: DataColumn<Geometry>
```

Comment:
I think the `T` is not needed, right? `ColumnsContainer<WithGeometry>` has the same effect

Reply:
Also, for the generated accessor we have both `ColumnsContainer` and `DataRow` extensions, so probably you should add `DataRow<WithGeometry>.geometry: Geometry` etc. too :)

Reply:
@koperagen could plugin help here, to generate all these extensions?

Reply:
do you mean the gradle plugin or the compiler plugin?

The compiler plugin is highly experimental still and it doesn't work yet in notebooks. So I wouldn't rely on that for now.

The gradle plugin, however, could generate these automatically based on the `@DataSchema` interfaces you defined. You'd have to add the dataframe gradle plugin to the geo module, just like we did in core, and make sure the generated files are part of sources to they are published along with the rest of the code.

Reply:
Sources generated by the plugin are automatically included. Makes sense to double check, but it's expected that you don't need to do anything extra. Just add this and it will generate required properties for annotated interfaces
plugins {
  id("org.jetbrains.kotlinx.dataframe")
 }

Reply:
Oops, forgot to add it, thank you!